ducktape/tests/test.py

```diff
@@ -64,7 +50,10 @@ def discover(self, base_dir, pattern=DEFAULT_TEST_FILE_PATTERN):
         - Discover test classes within each test module. A test class is a subclass of Test which is a leaf
           (i.e. it has no subclasses).
         """
-        test_files = self.find_test_files(base_dir, pattern)
+        if os.path.isfile(base_dir):
```

Comment:
This is exactly the type of thing that could use a unit test -- just a simple dummy test that you can run either by specifying the parent module or the file directly and make sure both do the right thing.